src/mindlessgen/qm/xtb.py

```diff
@@ -71,13 +90,33 @@ def optimize(
             # read the optimized molecule
             optimized_molecule = molecule.copy()
             optimized_molecule.read_xyz_from_file(temp_path / "xtbopt.xyz")
-
+            if np.any(np.isin(molecule.ati, get_lanthanides())):
+                # Reset the UHF value to the original value before returning the optimized molecule.
+                optimized_molecule.uhf = uhf_original
             return optimized_molecule
 
     def singlepoint(self, molecule: Molecule, verbosity: int = 1) -> str:
         """
         Perform a single-point calculation using xtb.
         """
+        if np.any(np.isin(molecule.ati, get_lanthanides())):
+            nel = 0
+            f_electrons = 0
+            for atom in molecule.ati:
+                nel += atom + 1
+                if atom in get_lanthanides():
+                    f_electrons += atom - 56
+            # Check if the number of the remaning electrons is even
+            test_rhf = nel - f_electrons - molecule.charge
+            if not test_rhf % 2 == 0:
+                raise SystemError(
+                    "Lanthanides detected, remaining number of electrons is not even."
+                )
```

Comment:
Very last comment: Can we put this part into a separate function `check_ligand_uhf` within the same file (module) so that we avoid copying code in `singlepoint` and `optimize`?

Reply:
It is now implemented